### ISSUE TITLE
Would be nice to have support for git ignore style patterns.

### ISSUE THREAD

**Issue by dave-kr** [davbeck/swift-glob#13]
Would be great if there was an option for patterns to follow the rules for git ignore style patterns. Used by `.gitignore` and GitHub `CODEOWNERS` files.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
https://git-scm.com/docs/gitignore#_pattern_format

**Comment by davbeck** [comment]
Agreed! A good first step for this would be to find (or create) a suite of tests, including failing cases and create a new `Pattern.Options` preset.

**Comment by davbeck** [comment]
Do you have a source for unit tests? Looking at the spec I think the default config should work.
